packages/math-input/src/components/tabbar/tabbar.tsx

```diff
@@ -19,41 +19,29 @@ const styles = StyleSheet.create({
     },
 });
 
-type TabbarState = {
-    selectedItem: number;
-};
-
 type Props = {
     items: ReadonlyArray<TabbarItemType>;
-    onSelect: (item: TabbarItemType) => void;
+    selectedItem: TabbarItemType;
+    onSelectItem: (item: TabbarItemType) => void;
 };
 
-class Tabbar extends React.Component<Props, TabbarState> {
-    state: TabbarState = {
-        selectedItem: 0,
-    };
-    render(): React.ReactNode {
-        const {items, onSelect} = this.props;
-        return (
-            <View style={styles.tabbar}>
-                {items.map((item, index) => (
-                    <TabbarItem
-                        key={`tabbar-item-${index}`}
-                        itemState={
-                            index === this.state.selectedItem
-                                ? "active"
-                                : "inactive"
-                        }
-                        itemType={item}
-                        onClick={() => {
-                            this.setState({selectedItem: index});
-                            onSelect(item);
-                        }}
-                    />
-                ))}
-            </View>
-        );
-    }
+function Tabbar(props: Props): React.ReactElement {
+    const {items, selectedItem, onSelectItem} = props;
```

Comment:
I personally like prop destructuring (obviously since I use it). I think it's very natural in functional components which tend to be small and more linear. Open to talking about it sometime though.

What I wish I could ban is prop spreading: `<SomeComponent {...props} />`